core/ledger/pvtdatastorage/store.go

```diff
@@ -476,6 +532,90 @@ func (s *Store) GetPvtDataByBlockNum(blockNum uint64, filter ledger.PvtNsCollFil
 	return blockPvtdata, nil
 }
 
+func (s *Store) retrieveLatestPurgeKeyCollMarkerHt(ns, coll string) (*version.Height, error) {
+	encVal, err := s.db.Get(
+		encodePurgeMarkerCollKey(
+			&purgeMarkerCollKey{
+				ns:   ns,
+				coll: coll,
+			},
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+	if encVal == nil {
+		return nil, nil
+	}
+	return decodePurgeMarkerVal(encVal)
+}
+
+func (s *Store) keyPotentiallyPurged(k *dataKey) (bool, error) {
+	purgeKeyCollMarkerHt, err := s.retrieveLatestPurgeKeyCollMarkerHt(k.ns, k.coll)
+	if purgeKeyCollMarkerHt == nil || err != nil {
+		return false, err
+	}
+
+	keyHt := &version.Height{
+		BlockNum: k.blkNum,
+		TxNum:    k.txNum,
+	}
+
+	return keyHt.Compare(purgeKeyCollMarkerHt) <= 0, nil
+}
+
+func (s *Store) removePurgedDataFromCollPvtRWset(k *dataKey, v *rwset.CollectionPvtReadWriteSet) error {
+	purgePossible, err := s.keyPotentiallyPurged(k)
+	if !purgePossible || err != nil {
+		return err
+	}
+
+	collRWSet, err := rwsetutil.CollPvtRwSetFromProtoMsg(v)
+	if err != nil {
+		return err
+	}
+
+	keyHt := &version.Height{
+		BlockNum: k.blkNum,
+		TxNum:    k.txNum,
+	}
+
+	filerInKVWrites := []*kvrwset.KVWrite{}
+	for _, w := range collRWSet.KvRwSet.Writes {
+		potentialPurgeMarker := encodePurgeMarkerKey(&purgeMarkerKey{
```

Comment:
```suggestion
		potentialPurgeMarkerKey := encodePurgeMarkerKey(&purgeMarkerKey{
```

Again, we won't remember what a 'potential' purge is. Could you add a comment explaining it.  Also the fact that we have a 'potential purge' at the collection level and a 'potential purge' at the hashed key level will confuse people reading the code. Maybe give at least one of them a more expressive name so that people can differentiate them.

Reply:
Did not understand this comment well but changed the variable name if that makes you happy